igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -55,24 +55,17 @@ class BuildController {
     Map getJobStatus(@PathVariable String master, @PathVariable Integer buildNumber, HttpServletRequest request) {
         def job = (String) request.getAttribute(
             HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE).split('/').drop(5).join('/')
-        if (buildMasters.filteredMap(BuildServiceProvider.JENKINS).containsKey(master)) {
-            Map result = objectMapper.convertValue(buildMasters.map[master].getBuild(job, buildNumber), Map)
+        if (buildMasters.map.containsKey(master)) {
+            Map result = objectMapper.convertValue(buildMasters.map[master].getGenericBuild(job, buildNumber), Map)
             try {
-                Map scm = objectMapper.convertValue(buildMasters.map[master].getGitDetails(job, buildNumber), Map)
-                if (scm?.action?.lastBuiltRevision?.branch?.name) {
-                    result.scm = scm?.action.lastBuiltRevision
-                    result.scm = result.scm.branch.collect {
-                        it.branch = it.name.split('/').last()
-                        it
-                    }
+                def scm = objectMapper.convertValue(buildMasters.map[master].getGenericGitRevision(job, buildNumber), Map)
+                if (scm) {
+                    result.scm = scm
```

Comment:
I tested this with
```
result.scm = objectMapper.convertValue(buildMasters.map[master].getGenericGitRevision(job, buildNumber), Map)
```

and got 
```
 ➜ curl http://localhost:8080/builds/status/1/dummy/build-my-project
{"building":false,"fullDisplayName":"build-my-project #1","number":1,"duration":52,"timestamp":"1456914302615","result":"SUCCESS","url":"http://54.194.40.92:8080/job/build-my-project/1/","scm":null}
```

I think this is because we are using the map type on this. I can probably return a GenericBuild from this endpoint and get the code nicer.